wire-ios-data-model/Source/Model/Message/ConversationMessage.swift

```diff
@@ -348,6 +350,31 @@ extension ZMMessage {
         return result
     }
 
+    @objc public var reactionDates: [String: Date] {
+        var result = [String: Date]()
+        for reaction in reactions where reaction.users.count > 0 {
+            result[reaction.unicodeValue!] = reaction.creationDate
+        }
+        return result
+    }
+
+    @objc public func reactionsSortedByCreationDate() -> [ReactionData] {
+        let reactionDates = self.reactionDates
+        return usersReaction.compactMap { reactionString, users in
+            return ReactionData(
+                reactionString: reactionString,
+                users: users,
+                creationDate: reactionDates[reactionString] ?? Date.distantPast
+            )
+        }.sorted {
+            if $0.creationDate == $1.creationDate {
+                return $0.users.count > $1.users.count
```

Comment:
Is this really only for all the old reactions that have a distant past date?

Reply:
Not really, it may happen that two reactions have the same date if more than one new reaction arrives at the same time from other user.

Reply:
Then I'd avoid sorting on count since it won't be a stable order. If we sort on the actual string value would be.